modules/base/scale/timelinescale.cpp

```diff
@@ -0,0 +1,120 @@
+ * Copyright (c) 2014-2025                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/base/scale/timelinescale.h>
+
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/scene/scene.h>
+#include <openspace/util/updatestructures.h>
+#include <openspace/util/time.h>
+#include <optional>
+
+namespace {
+    constexpr openspace::properties::Property::PropertyInfo ShouldInterpolateInfo = {
+        "ShouldInterpolate",
+        "Should Interpolate",
+        "If this value is set to 'true', an interpolation is applied between the given "
+        "keyframes. If this value is set to 'false', the interpolation is not applied.",
+        openspace::properties::Property::Visibility::AdvancedUser
+    };
+
+    // This `Scale` uses a timeline of other `Scale` classes to calculate the final
+    // scale factor for the attached scene graph node. The current in-game time is used to
+    // determine which specific keyframe to currently use. It is also possible to
+    // interpolate between two adjacent keyframes.
+    struct [[codegen::Dictionary(TimelineScale)]] Parameters {
+        // A table of keyframes, with keys formatted as YYYY-MM-DDTHH:MM:SS and values
+        // that are valid Scale objects
```

Comment:
```suggestion
        // that are valid Scale objects.
```

I guess this should be in the other PR, though?